modules/jpa/src/main/java/io/oasp/module/jpa/common/api/ApplicationDao.java

```diff
@@ -0,0 +1,14 @@
+package io.oasp.module.jpa.common.api;
+
+import io.oasp.module.jpa.dataaccess.api.Dao;
+import io.oasp.module.jpa.dataaccess.api.GenericRevisionedDao;
+import io.oasp.module.jpa.dataaccess.api.MutablePersistenceEntity;
+
+/**
+ * Interface for all {@link GenericRevisionedDao DAOs} (Data Access Object) of this application.
+ *
+ * @param <ENTITY> is the type of the managed entity.
+ */
+public interface ApplicationDao<ENTITY extends MutablePersistenceEntity<Long>> extends Dao<ENTITY> {
```

Comment:
ApplicationDao is usually the one that goes to the application in order to let the project add its own bells and whistles there. Having this in our module would IMHO cause namespace issues with CobiGen that generates another `ApplicationDao` inside the app and then cannot resolve the imports automatically anymore as the name is ambiguous.

Reply:
As LegacyApplicationDaoImpl  is implementing it , its name can be changed to LegacyApplicationDao